### ISSUE TITLE
Error parsing statement

### ISSUE THREAD

**Issue by tinygrox** [KSP-RO/ContractConfigurator#39]
## Game Version: 
1.12.5
## Mods Installed: 
- ModuleManager 
- ContractConfigurator
- Contract Pack: Field Research
## Problem
My friends trying to translate the Contract Pack **Field Research** and he doing this in _GameData\ContractPacks\FieldResearch\Experiment.cfg_ :

```
CONTRACT_TYPE
{
    name = FS_Experiment
    ...
    // title = Perform @experiment.Name().ToLower().FirstCap() experiments on @targetBody
    title = 在 @targetBody 上进行 @experiment.Name().ToLower().FirstCap() 实验。
   ...
}
```

when he enter the game, the contract **will just gone**

I look into the **KSP.log** and found:
```bash
ContractConfigurator.ContractType: CONTRACT_TYPE 'FS_Experiment': Error parsing title

ArgumentException: Expected a valid expression, found: '实'
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseToken () (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseMethod[TResult] (ContractConfigurator.ExpressionParser.BaseParser+Token token, T obj, System.Boolean isFunction) (at <20a3a039060941bfb465629683bb64bf>:0)
	System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <9577ac7a62ef43179789031239ba8798>:0)
	Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
	System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <9577ac7a62ef43179789031239ba8798>:0)
	System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at <9577ac7a62ef43179789031239ba8798>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseMethod[TResult] (ContractConfigurator.ExpressionParser.BaseParser+Token token, T obj, System.Boolean isFunction) (at <20a3a039060941bfb465629683bb64bf>:0)
	System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <9577ac7a62ef43179789031239ba8798>:0)
	Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
	System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <9577ac7a62ef43179789031239ba8798>:0)
	System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at <9577ac7a62ef43179789031239ba8798>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseMethod[TResult] (ContractConfigurator.ExpressionParser.BaseParser+Token token, T obj, System.Boolean isFunction) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].CompleteIdentifierParsing[U] (U value) (at <20a3a039060941bfb465629683bb64bf>:0)
	System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <9577ac7a62ef43179789031239ba8798>:0)
	Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
	System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <9577ac7a62ef43179789031239ba8798>:0)
	System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at <9577ac7a62ef43179789031239ba8798>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseSpecialIdentifier (ContractConfigurator.ExpressionParser.BaseParser+Token token) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.StringExpressionParser.ParseStatement[TResult] () (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseExpression (System.String key, System.String expression, ContractConfigurator.ExpressionParser.DataNode dataNode) (at <20a3a039060941bfb465629683bb64bf>:0)
	Rethrow as Exception: Error parsing statement.
	Error occurred near '*':
	在 @targetBody 上进行 @experiment.Name().ToLower().FirstCap() 实验。
	..........................................................* <-- HERE
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseExpression (System.String key, System.String expression, ContractConfigurator.ExpressionParser.DataNode dataNode) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ConfigNodeUtil.ParseSingleValue[T] (System.String key, System.String stringValue, System.Boolean allowExpression) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ConfigNodeUtil.ParseValue[T] (ConfigNode configNode, System.String key, System.Boolean allowExpression) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ConfigNodeUtil.ParseValue[T] (ConfigNode configNode, System.String key, System.Action`1[T] setter, ContractConfigurator.IContractConfiguratorFactory obj, T defaultValue, System.Func`2[T,TResult] validation) (at <20a3a039060941bfb465629683bb64bf>:0)
	UnityEngine.DebugLogHandler:LogException(Exception, Object)
	ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
	UnityEngine.Debug:LogException(Exception)
	ContractConfigurator.LoggingUtil:LogException(Exception)
	ContractConfigurator.ConfigNodeUtil:ParseValue(ConfigNode, String, Action`1, IContractConfiguratorFactory, String, Func`2)
	ContractConfigurator.ConfigNodeUtil:ParseValue(ConfigNode, String, Action`1, IContractConfiguratorFactory, Func`2)
	ContractConfigurator.DeferredLoadUtil:ExecuteLoad(DeferredLoadObject`1)
	System.Reflection.MethodBase:Invoke(Object, Object[])
	ContractConfigurator.ConfigNodeUtil:ExecuteDeferredLoads()
	ContractConfigurator.ContractType:Load(ConfigNode)
	ContractConfigurator.<LoadContractTypeConfig>d__32:MoveNext()
	ContractConfigurator.<FinalizeContractTypeLoad>d__31:MoveNext()
	UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
ContractConfigurator.ContractType: Errors encountered while trying to load CONTRACT_TYPE 'FS_Experiment'
```

And then I delete the `实验` like this:
```
CONTRACT_TYPE
{
    name = FS_Experiment
    ...
    // title = Perform @experiment.Name().ToLower().FirstCap() experiments on @targetBody
    title = 在 @targetBody 上进行 @experiment.Name().ToLower().FirstCap()
   ...
}
```
**it works again.** The contract is back in Mission Control.

**Comment by siimav** [comment]
Honestly no idea what's going on. Maybe try wrapping the expression between `()`. Or try extracting the expression as a `DATA` node.

**Comment by zhangyuesai** [comment]
After a few tests I found out that **this error occurs when there's an expression and a non-ASCII character after that expression.**

# Issue
_Sometimes_ an `Error parsing statement` occurs when there's a non-ASCII character after a expression. This does not happen to every contract pack and expression, but if it happens it could be triggered by many non-ASCII characters (eg. Chinese/Korean characters, accented latin letters, non-ASCII symbols), not just Chinese characters.

# Dependencies
- KSP 1.12.5
- Module Manager 4.2.3
- Contract Configurator 2.9.1.0
- Contract Pack: Field Research 1.2.2
- Contract Pack: Bases and Stations Reborn 3.7.3.1
- Contract Pack: Rover Missions 0.1.7.0

# Problem
I was translating some contract packs to Chinese. When I tried replacing some English texts with Chinese characters, Contract Configurator would not properly parse the statement and the corresponding contract would not be loaded into the game. Below are A few examples.

If I try these:

In `GameData/ContractPacks/FieldResearch/Experiment.cfg`:
```
CONTRACT_TYPE
{
    name = FS_Experiment
...
    // title = Perform @experiment.Name().ToLower().FirstCap() experiments on @targetBody
    title = Perform @experiment.Name().ToLower().FirstCap() 中文 @targetBody
...
}
```

In `GameData/ContractPacks/KerbinSpaceStation/MaintenanceMissions/Evacuate.cfg`:
```
CONTRACT_TYPE
{
    name = Evacuate
...
        PARAMETER
        {
            ...
            // title = Leave no Kerbals on board @/targetVessel1.
            title = @/targetVessel1 ↑ 一名坎巴拉人也不要留下
        }
...
}
```

In `GameData/ContractPacks/KerbinSpaceStation/MaintenanceMissions/PayloadSpecialist.cfg`:
```
CONTRACT_TYPE
{
    name = PayloadSpecialist
...
    PARAMETER
    {
        ...
        // completionText = The @/brokenPart has been repaired!
        completionText = @/brokenPart 한국어
        ...
    }
...
}
```

In `GameData/ContractPacks/Spacetux/RoverMissions/IteratorRover.cfg`:
```
CONTRACT_TYPE
{
    name = RoverExplore
...
                PARAMETER
                {
                    ...
                    // title = Scientist @/kerbals.ElementAt(@index - 1) would like you to go investigate @/currentLocs.ElementAt(@index - 1)
                    title = 科学家 @/kerbals.ElementAt(@index - 1) á @/currentLocs.ElementAt(@index - 1)
                }
...
}
```
An `Error parsing statement` will occur, causing Contract Configurator fail to load the corresponding cntract. The `KSP.log` reads:
```
[LOG 12:10:22.714] [INFO] ContractConfigurator.ContractType: Loading CONTRACT_TYPE: 'FS_Experiment'
[ERR 12:10:22.730] ContractConfigurator.ContractType: CONTRACT_TYPE 'FS_Experiment': Error parsing title

[EXC 12:10:22.734] ArgumentException: Expected a valid expression, found: '中'
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseToken () (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseMethod[TResult] (ContractConfigurator.ExpressionParser.BaseParser+Token token, T obj, System.Boolean isFunction) (at <20a3a039060941bfb465629683bb64bf>:0)
	System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <9577ac7a62ef43179789031239ba8798>:0)
	Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
	System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <9577ac7a62ef43179789031239ba8798>:0)
	System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at <9577ac7a62ef43179789031239ba8798>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseMethod[TResult] (ContractConfigurator.ExpressionParser.BaseParser+Token token, T obj, System.Boolean isFunction) (at <20a3a039060941bfb465629683bb64bf>:0)
	System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <9577ac7a62ef43179789031239ba8798>:0)
	Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
	System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <9577ac7a62ef43179789031239ba8798>:0)
	System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at <9577ac7a62ef43179789031239ba8798>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseMethod[TResult] (ContractConfigurator.ExpressionParser.BaseParser+Token token, T obj, System.Boolean isFunction) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].CompleteIdentifierParsing[U] (U value) (at <20a3a039060941bfb465629683bb64bf>:0)
	System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <9577ac7a62ef43179789031239ba8798>:0)
	Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
	System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <9577ac7a62ef43179789031239ba8798>:0)
	System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at <9577ac7a62ef43179789031239ba8798>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseSpecialIdentifier (ContractConfigurator.ExpressionParser.BaseParser+Token token) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.StringExpressionParser.ParseStatement[TResult] () (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseExpression (System.String key, System.String expression, ContractConfigurator.ExpressionParser.DataNode dataNode) (at <20a3a039060941bfb465629683bb64bf>:0)
	Rethrow as Exception: Error parsing statement.
	Error occurred near '*':
	Perform @experiment.Name().ToLower().FirstCap() 中文 @targetBody
	................................................* <-- HERE
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseExpression (System.String key, System.String expression, ContractConfigurator.ExpressionParser.DataNode dataNode) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ConfigNodeUtil.ParseSingleValue[T] (System.String key, System.String stringValue, System.Boolean allowExpression) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ConfigNodeUtil.ParseValue[T] (ConfigNode configNode, System.String key, System.Boolean allowExpression) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ConfigNodeUtil.ParseValue[T] (ConfigNode configNode, System.String key, System.Action`1[T] setter, ContractConfigurator.IContractConfiguratorFactory obj, T defaultValue, System.Func`2[T,TResult] validation) (at <20a3a039060941bfb465629683bb64bf>:0)
	UnityEngine.DebugLogHandler:LogException(Exception, Object)
	ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
	UnityEngine.Debug:LogException(Exception)
	ContractConfigurator.LoggingUtil:LogException(Exception)
	ContractConfigurator.ConfigNodeUtil:ParseValue(ConfigNode, String, Action`1, IContractConfiguratorFactory, String, Func`2)
	ContractConfigurator.ConfigNodeUtil:ParseValue(ConfigNode, String, Action`1, IContractConfiguratorFactory, Func`2)
	ContractConfigurator.DeferredLoadUtil:ExecuteLoad(DeferredLoadObject`1)
	System.Reflection.MethodBase:Invoke(Object, Object[])
	ContractConfigurator.ConfigNodeUtil:ExecuteDeferredLoads()
	ContractConfigurator.ContractType:Load(ConfigNode)
	ContractConfigurator.<LoadContractTypeConfig>d__32:MoveNext()
	ContractConfigurator.<FinalizeContractTypeLoad>d__31:MoveNext()
	UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
[WRN 12:10:22.736] ContractConfigurator.ContractType: Errors encountered while trying to load CONTRACT_TYPE 'FS_Experiment'

...

[ERR 12:10:22.989] ContractConfigurator.HasCrewFactory: CONTRACT_TYPE 'Evacuate', PARAMETER 'HasCrew' of type 'HasCrew': Error parsing title

[EXC 12:10:22.991] ArgumentException: Expected a valid expression, found: '↑'
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseToken () (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseStatement[TResult] (T lval) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseStatementInner[TResult] () (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseStatement[TResult] () (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.StringExpressionParser.ParseStatement[TResult] () (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseExpression (System.String key, System.String expression, ContractConfigurator.ExpressionParser.DataNode dataNode) (at <20a3a039060941bfb465629683bb64bf>:0)
	Rethrow as Exception: Error parsing statement.
	Error occurred near '*':
	@/targetVessel1 ↑ 一名坎巴拉人也不要留下
	................* <-- HERE
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseExpression (System.String key, System.String expression, ContractConfigurator.ExpressionParser.DataNode dataNode) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ConfigNodeUtil.ParseSingleValue[T] (System.String key, System.String stringValue, System.Boolean allowExpression) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ConfigNodeUtil.ParseValue[T] (ConfigNode configNode, System.String key, System.Boolean allowExpression) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ConfigNodeUtil.ParseValue[T] (ConfigNode configNode, System.String key, System.Action`1[T] setter, ContractConfigurator.IContractConfiguratorFactory obj, T defaultValue, System.Func`2[T,TResult] validation) (at <20a3a039060941bfb465629683bb64bf>:0)
	UnityEngine.DebugLogHandler:LogException(Exception, Object)
	ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
	UnityEngine.Debug:LogException(Exception)
	ContractConfigurator.LoggingUtil:LogException(Exception)
	ContractConfigurator.ConfigNodeUtil:ParseValue(ConfigNode, String, Action`1, IContractConfiguratorFactory, String, Func`2)
	ContractConfigurator.ConfigNodeUtil:ParseValue(ConfigNode, String, Action`1, IContractConfiguratorFactory, String)
	ContractConfigurator.ParameterFactory:Load(ConfigNode)
	ContractConfigurator.HasCrewFactory:Load(ConfigNode)
	ContractConfigurator.ParameterFactory:GenerateParameterFactory(ConfigNode, ContractType, ParameterFactory&, ParameterFactory)
	ContractConfigurator.ParameterFactory:GenerateParameterFactory(ConfigNode, ContractType, ParameterFactory&, ParameterFactory)
	ContractConfigurator.ContractType:Load(ConfigNode)
	ContractConfigurator.<LoadContractTypeConfig>d__32:MoveNext()
	ContractConfigurator.<FinalizeContractTypeLoad>d__31:MoveNext()
	UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
[WRN 12:10:22.995] ContractConfigurator.ContractType: Errors encountered while trying to load CONTRACT_TYPE 'Evacuate'

...

[ERR 12:10:23.023] ContractConfigurator.DurationFactory: CONTRACT_TYPE 'PayloadSpecialist', PARAMETER 'Duration' of type 'Duration': Error parsing completionText

[EXC 12:10:23.024] ArgumentException: Expected a valid expression, found: '한'
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseToken () (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseStatement[TResult] (T lval) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseStatementInner[TResult] () (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseStatement[TResult] () (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.StringExpressionParser.ParseStatement[TResult] () (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseExpression (System.String key, System.String expression, ContractConfigurator.ExpressionParser.DataNode dataNode) (at <20a3a039060941bfb465629683bb64bf>:0)
	Rethrow as Exception: Error parsing statement.
	Error occurred near '*':
	@/brokenPart 한국어
	.............* <-- HERE
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseExpression (System.String key, System.String expression, ContractConfigurator.ExpressionParser.DataNode dataNode) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ConfigNodeUtil.ParseSingleValue[T] (System.String key, System.String stringValue, System.Boolean allowExpression) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ConfigNodeUtil.ParseValue[T] (ConfigNode configNode, System.String key, System.Boolean allowExpression) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ConfigNodeUtil.ParseValue[T] (ConfigNode configNode, System.String key, System.Action`1[T] setter, ContractConfigurator.IContractConfiguratorFactory obj, T defaultValue, System.Func`2[T,TResult] validation) (at <20a3a039060941bfb465629683bb64bf>:0)
	UnityEngine.DebugLogHandler:LogException(Exception, Object)
	ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
	UnityEngine.Debug:LogException(Exception)
	ContractConfigurator.LoggingUtil:LogException(Exception)
	ContractConfigurator.ConfigNodeUtil:ParseValue(ConfigNode, String, Action`1, IContractConfiguratorFactory, String, Func`2)
	ContractConfigurator.ConfigNodeUtil:ParseValue(ConfigNode, String, Action`1, IContractConfiguratorFactory, String)
	ContractConfigurator.DurationFactory:Load(ConfigNode)
	ContractConfigurator.ParameterFactory:GenerateParameterFactory(ConfigNode, ContractType, ParameterFactory&, ParameterFactory)
	ContractConfigurator.ContractType:Load(ConfigNode)
	ContractConfigurator.<LoadContractTypeConfig>d__32:MoveNext()
	ContractConfigurator.<FinalizeContractTypeLoad>d__31:MoveNext()
	UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
[WRN 12:10:23.027] ContractConfigurator.ContractType: Errors encountered while trying to load CONTRACT_TYPE 'PayloadSpecialist'

...

[ERR 12:10:23.186] ContractConfigurator.VisitWaypointFactory: CONTRACT_TYPE 'RoverExplore', PARAMETER 'LocationArrival_@index' of type 'VisitWaypoint': Error parsing title

[EXC 12:10:23.189] ArgumentException: Expected a valid expression, found: 'á'
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseToken () (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseMethod[TResult] (ContractConfigurator.ExpressionParser.BaseParser+Token token, T obj, System.Boolean isFunction) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ListExpressionParser`1[T].ParseMethod[TResult] (ContractConfigurator.ExpressionParser.BaseParser+Token token, System.Collections.Generic.List`1[T] obj, System.Boolean isFunction) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].CompleteIdentifierParsing[U] (U value) (at <20a3a039060941bfb465629683bb64bf>:0)
	System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <9577ac7a62ef43179789031239ba8798>:0)
	Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
	System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <9577ac7a62ef43179789031239ba8798>:0)
	System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at <9577ac7a62ef43179789031239ba8798>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseSpecialIdentifier (ContractConfigurator.ExpressionParser.BaseParser+Token token) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.StringExpressionParser.ParseStatement[TResult] () (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseExpression (System.String key, System.String expression, ContractConfigurator.ExpressionParser.DataNode dataNode) (at <20a3a039060941bfb465629683bb64bf>:0)
	Rethrow as Exception: Error parsing statement.
	Error occurred near '*':
	科学家 @/kerbals.ElementAt(@index - 1) á @/currentLocs.ElementAt(@index - 1)
	....................................* <-- HERE
	ContractConfigurator.ExpressionParser.ExpressionParser`1[T].ParseExpression (System.String key, System.String expression, ContractConfigurator.ExpressionParser.DataNode dataNode) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ConfigNodeUtil.ParseSingleValue[T] (System.String key, System.String stringValue, System.Boolean allowExpression) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ConfigNodeUtil.ParseValue[T] (ConfigNode configNode, System.String key, System.Boolean allowExpression) (at <20a3a039060941bfb465629683bb64bf>:0)
	ContractConfigurator.ConfigNodeUtil.ParseValue[T] (ConfigNode configNode, System.String key, System.Action`1[T] setter, ContractConfigurator.IContractConfiguratorFactory obj, T defaultValue, System.Func`2[T,TResult] validation) (at <20a3a039060941bfb465629683bb64bf>:0)
	UnityEngine.DebugLogHandler:LogException(Exception, Object)
	ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
	UnityEngine.Debug:LogException(Exception)
	ContractConfigurator.LoggingUtil:LogException(Exception)
	ContractConfigurator.ConfigNodeUtil:ParseValue(ConfigNode, String, Action`1, IContractConfiguratorFactory, String, Func`2)
	ContractConfigurator.ConfigNodeUtil:ParseValue(ConfigNode, String, Action`1, IContractConfiguratorFactory, Func`2)
	ContractConfigurator.DeferredLoadUtil:ExecuteLoad(DeferredLoadObject`1)
	System.Reflection.MethodBase:Invoke(Object, Object[])
	ContractConfigurator.ConfigNodeUtil:ExecuteDeferredLoads()
	ContractConfigurator.ContractType:Load(ConfigNode)
	ContractConfigurator.<LoadContractTypeConfig>d__32:MoveNext()
	ContractConfigurator.<FinalizeContractTypeLoad>d__31:MoveNext()
	UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
[WRN 12:10:23.196] ContractConfigurator.ContractType: Errors encountered while trying to load CONTRACT_TYPE 'RoverExplore'
```
@siimav , your solution of wrapping the expressions between `()` seems work. Thanks for that! However, this bug doesn't only affect Chinese translators，but also other languages (Frensh, Spanish, etc.) and modders who use non-ASCII characters in their contracts. So it may still be worth looking into. 

Thanks!

**Comment by tinygrox** [comment]
@siimav I've found the location of the problem, I guess.
https://github.com/KSP-RO/ContractConfigurator/blob/ea17ca500efb863540e915bc91ae113f36bafc09/source/ContractConfigurator/ExpressionParser/ExpressionParser.cs#L732-L735
If I change the code to this:
```C#
if (char.IsLetter(c))
{
    return ParseIdentifier();
}
```
the problem seems to be resolved, but I haven't tested it in other languages yet.

**Comment by siimav** [comment]
@tinygrox To be clear - you or someone you know made the change above; compiled it onto a DLL; and then actually tested ingame to make sure everything works correctly?

**Comment by tinygrox** [comment]
> @tinygrox To be clear - you or someone you know made the change above; compiled it onto a DLL; and then actually tested ingame to make sure everything works correctly?

I made the changes and compiled them into a DLL file. I've conducted some testing in the game, but it wasn't comprehensive. The changes seem to work well, but I haven't tested every scenario. At least the Chinese and English are OK. I mean that every contract from _**Contract Pack: Field Research**_ appears in Mission Control, there are no errors in **_KSP.log_**.
So, let's make it clear, I **can't guarantee** that everything works correctly.
